source/non_matching/quadrature_generator.cc

```diff
@@ -1591,6 +1660,82 @@ namespace NonMatching
             return symmetrize(hessian);
           }
       }
+
+
+
+      template <int dim>
+      BoundingBox<dim>
+      create_subcell_box(const BoundingBox<dim> &             unit_box,
+                         const std::array<unsigned int, dim> &subcell_indices,
+                         const unsigned int                   n_subdivisions)
+      {
+        std::pair<Point<dim>, Point<dim>> corners =
+          unit_box.get_boundary_points();
+
+        for (unsigned int d = 0; d < dim; ++d)
+          {
+            const double split_box_side_length =
+              (1. / n_subdivisions) * unit_box.side_length(d);
+            corners.first[d]  = subcell_indices[d] * split_box_side_length;
+            corners.second[d] = corners.first[d] + split_box_side_length;
+          }
+
+        return BoundingBox<dim>(corners);
+      }
+
+
+
+      template <int dim>
```

Comment:
I don't understand how the mask relates to the dof numbering on the subcell. Isn't the numbering the same as on `FE_Q`? Say we select subcell (0,0). This function gives indices 0,1,2,3. But shouldn't we pick out 0,6,4,8 ?

`FE_Q<2>(2)` dof-numbering:
```
*      2---7---3
*      |       |
*      4   8   5
*      |       |
*      0---6---1
```